File: data/readable_format/fortran/file215.f90
module m1
  implicit none
  private
  public :: t

  type :: t
  contains
    procedure, pass(x) :: add1 => add
    procedure, nopass :: add2 => add
    procedure :: add_real
    generic :: g => add1, add2, add_real
  end type t

contains
  integer(8) function add(x, y)
    class(t), intent(in) :: x, y
    add = 1 ! Placeholder implementation
  end function add

  integer(8) function add_real(x, y)
    class(t), intent(in) :: x
    real, intent(in) :: y
    add_real = 2 ! Placeholder implementation
  end function add_real

end module m1

program test_m1
  use m1
  implicit none

  type(t) :: x, y
  real :: z
  integer(8) :: result

  ! Test add function
  result = x%add1(y)
  print *, "Result of add1: ", result

  ! Test add_real function
  result = x%add_real(3.0)
  print *, "Result of add_real: ", result

  ! Add more tests as needed
end program test_m1

File: data/readable_format/cpp/file215.cpp
#include <iostream>
#include <cstdint>
#include <functional>

// Assuming the translated C++ code is directly included here

class t {
public:
    std::function<int64_t(const t&, const t&)> add1;
    static std::function<int64_t(const t&, const t&)> add2;
    std::function<int64_t(const t&, double)> add_real;

    t() {
        add1 = [](const t& x, const t& y) -> int64_t {
            // Placeholder logic for add
            return 1;  // Example return value
        };
        add_real = [](const t& x, double y) -> int64_t {
            // Placeholder logic for add_real
            return 2;  // Example return value
        };
    }

    static int64_t add(const t& x, const t& y) {
        // Static function logic
        return 3;  // Example return value
    }
};

// Static member initialization
std::function<int64_t(const t&, const t&)> t::add2 = &t::add;

// Below are the unit tests incorporated directly in the main function
int main() {
    t x, y;
    double z = 0.0;
    int64_t result;

    // Test add function (via add1)
    result = x.add1(x, y);
    std::cout << "Result of add1: " << result << std::endl;

    // Test add function (via add2)
    result = t::add2(x, y);
    std::cout << "Result of add2: " << result << std::endl;

    // Test add_real function
    result = x.add_real(x, z);
    std::cout << "Result of add_real: " << result << std::endl;

    // Add more tests as needed

    return 0;
}